civis/ml/_model.py

```diff
@@ -1146,6 +1148,8 @@ def from_existing(cls, train_job_id, train_run_id='latest', client=None):
         klass.train_result_ = fut
 
         # Set prediction template corresponding to training template
+        # The bug is here: We expect a job that is based off of a Training template
+        # but it is really based off of a Registration template
```

Comment:
`from_existing` looks correct to me, because this class method _can_ be used on its own (so not necessarily called by `register_pretrained_model`) and it does expect a training job ID from its function signature in order to infer the training template ID here. If there's a bug, it might be at your other comment above at `register_pretrained_model`. I'll chat with you offline to look at this together more closely.

Reply:
Yea, you're right. I think the issue is further up. What do you recommend as the fix to the problem?